File: app/src/main/java/com/zack/enderplan/view/contract/AllTypesViewContract.java
package com.zack.enderplan.view.contract;

import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.View;

import com.zack.enderplan.view.adapter.TypeListAdapter;

public interface AllTypesViewContract extends BaseViewContract {

    void showInitialView(TypeListAdapter typeListAdapter, ItemTouchHelper itemTouchHelper);

    void onTypeItemClicked(int position, View typeItem);

    void onTypeCreated(int scrollTo);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/AllTypesViewContract.kt
package net.zackzhang.app.end.view.contract

import android.support.v7.widget.helper.ItemTouchHelper
import android.view.View

import net.zackzhang.app.end.view.adapter.TypeListAdapter

interface AllTypesViewContract : BaseViewContract {

    fun showInitialView(typeListAdapter: TypeListAdapter, itemTouchHelper: ItemTouchHelper)

    fun onTypeItemClicked(position: Int, typeItem: View)

    fun onTypeCreated(scrollTo: Int)
}
